sort/bucketsort.go

```diff
@@ -0,0 +1,46 @@
+package sort
+
+import "github.com/TheAlgorithms/Go/constraints"
+
+// sorting an array using buckets
+// mainly useful when input is uniformly distributed over a range
+func BucketSort[T constraints.Number](arr []T) []T {
```

Comment:
```suggestion
func Bucket[T constraints.Number](arr []T) []T {
```

Bucket is in the name of the package. When we would call it, we would use something like `sort.Bucket` which is more than enough indication that this is bucket sort.